crates/binjs_io/src/binjs_json/write.rs

```diff
@@ -112,24 +111,30 @@ impl TokenWriter for TreeTokenWriter {
                 interface: tag,
                 field_values,
             }) => {
-                let mut obj = json::object::Object::with_capacity(2);
-                obj.insert("@TYPE", JSON::String("tagged tuple".to_string()));
-                obj.insert("@INTERFACE", JSON::String(tag.as_str().to_string()));
+                let mut obj = serde_json::Map::with_capacity(2);
```

Comment:
Couldn't this be

```rust
let mut obj = json!{
  "@TYPE": "tagged tuple",
  "@INTERFACE": tag
};
```
?

Also in a few places below.

Reply:
It could! But as I mentioned in the description, I intentionally tried to make this 1:1 replacement without refactorings for easier reviews and validation :)

Reply:
Sounds good :)